orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py

```diff
@@ -1210,3 +1209,63 @@ def forward(self, model_input):
     assert not can_run
 
     del os.environ["ORTMODULE_SKIPPED_AUTOGRAD_FUNCTIONS"]
+
+
+def test_pythonop_training_mode():
+    def check_pythonop_training_mode(model, is_eval_mode):
+        ## make sure the ort's PythonOp's training_mode is correct
+        if is_eval_mode:
+            onnx_nodes = (
+                model._torch_module._execution_manager._inference_manager._onnx_models.exported_model.graph.node
+            )
+        else:
+            onnx_nodes = model._torch_module._execution_manager._training_manager._onnx_models.exported_model.graph.node
+
+        found_pythonop = False
+        for node in onnx_nodes:
+            if node.op_type == "PythonOp":
+                found_pythonop = True
+                for attr in node.attribute:
+                    if attr.name == "training_mode":
+                        if is_eval_mode:
+                            assert attr.i == 0, f"in eval mode, it shoule be 0, while it is {attr.i} now"
+                        else:
+                            assert attr.i == 1, f"in training mode, it should be 1, while it is {attr.i} now"
+
+        assert found_pythonop, "PythonOp should be found in the exported model"
+
+    class TestFunction(torch.autograd.Function):
+        @staticmethod
+        # bias is an optional argument
+        def forward(ctx, x):
+            ctx.save_for_backward(x)
+            return x * 0.5 * (1.0 + torch.tanh(0.79788456 * x * (1 + 0.044715 * x * x)))
+
+        @staticmethod
+        def backward(ctx, grad_output):
+            x = ctx.saved_tensors
```

Comment:
nit: maybe you can just let compute using x and grad_output, otherwise, the pylint complains unused variables.